### PR TITLE
Update databricks deployment docs

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
[Kedro starters #119](https://github.com/kedro-org/kedro-starters/pull/119) introduces a specific environment for deploying the pyspark-iris starter on databricks. This PR makes the relevant changes to the documentation. Additionally, some of the specifications and images were out of date, and are updated in this PR.

Also resolves https://github.com/kedro-org/kedro/issues/2203.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by jmholzer** [comment]
This is great work, thank you! You actually addressed the problems in https://github.com/kedro-org/kedro/issues/2203 also 🙂 kudos for noticing.

I'll review this properly, though it looks good to me so far.

File: docs/source/deployment/databricks.md [hunk 1]
```diff
@@ -60,22 +60,20 @@ You should get a similar output:
...
                     INFO     Pipeline execution completed successfully.                                      runner.py:89
 ```
 ### 3. Create a Databricks cluster
 
-If you already have an active cluster with runtime version `7.1`, you can skip this step. Here is [how to find clusters in your Databricks workspace](https://docs.databricks.com/clusters/clusters-manage.html).
+If you already have an active cluster with runtime version `7.3`, you can skip this step. Here is [how to find clusters in your Databricks workspace](https://docs.databricks.com/clusters/clusters-manage.html).
 
 Follow the [Databricks official guide to create a new cluster](https://docs.databricks.com/clusters/create-cluster.html). For the purpose of this tutorial (and to minimise costs) we recommend the following settings:
-* Runtime: `7.1 (Scala 2.12, Spark 3.0.0)`
+* Runtime: `7.3 (Scala 2.12, Spark 3.0.1)`
```
Comment (jmholzer) [inline docs/source/deployment/databricks.md:71]: ✅ I think this is the best version to specify for now, though we may have to revisit this when we rewrite as part of https://github.com/kedro-org/kedro/issues/2185.

**Review by jmholzer — APPROVED** [review]
Awesome work! Thank you 🎉

**Review by stichbury — APPROVED** [review]
LGTM 🌟

**Pull request merged**